### PR TITLE
Refactor schema and seed to produce realistic data

### PR DESCRIPTION
**Pull request by matt-leyrat**
**Main changes: **
- item.colors is now an array of objects, each with a colorName and an array of size objects
- size objects within each element of item.colors have a stock property (random number < 5)
- when a color is picked and the dropdown is clicked, sizes that are out of stock now display (out of stock)
- designedFor and fit now live in the same object: item.features, and are dynamically rendered in the accordion

### CODE REVIEW

File: client/src/components/AccordionTab.jsx [hunk 1]
```diff
@@ -32,7 +32,13 @@ const Tab = (props) => {
       <TabContainer>
         <TabTitle onClick={props.updateTabDisplay}> {props.title}</TabTitle>
         <TabContentContainer >
-          <AccordionTabContent care={props.care} tabTitle={props.title} tabContent={props.content} />
+          <AccordionTabContent
+          fabric={props.fabric}
+          care={props.care}
+          tabTitle={props.title}
+          tabContent={props.content}
+          features={props.features}
```
Comment (garrowat) [inline client/src/components/AccordionTab.jsx]: I'm a fan of a little destructuring here to eliminate the reuse of the props object.

File: client/src/components/AccordionTabContent.jsx [hunk 2]
```diff
@@ -61,7 +99,20 @@ const AccordionTabContent = (props) => {
...
+        </ItemFeature>
+        <ItemFeature>
+          <ItemFeatureName>
+            {props.features.fit.join(', ').toUpperCase()}
+          {/* {props.features.fit.map((fit) => {
+            return fit
+          })} */}
+          </ItemFeatureName>
+        </ItemFeature>
+      </ItemFeaturesContainer>
```
Comment (garrowat) [inline client/src/components/AccordionTabContent.jsx]: This all looks good to me, nice looking components.

File: client/src/components/AccordionTab.jsx (hunk 1 above)
Comment (matt-leyrat) [reply client/src/components/AccordionTab.jsx]: Yes! I need to make that a habit ASAP 👍

File: client/src/components/AccordionTabContent.jsx (hunk 2 above)
Comment (matt-leyrat) [reply client/src/components/AccordionTabContent.jsx]: Thanks Garrett!

File: client/src/components/ItemSummary.jsx [hunk 3]
```diff
@@ -36,34 +36,37 @@ class ItemSummary extends React.Component {
   }
 
   selectColor(event) {
-    console.log(event.target.value);
-    this.setState({ selectedColor: event.target.value })
+    var capitalizedColor = event.target.value.charAt(0).toUpperCase() + event.target.value.slice(1);
+    var parsedColors = JSON.parse(this.state.item[0].colors);
+    var matchingColorObj = parsedColors.find(color => color.colorName === capitalizedColor);
```
Comment (garrowat) [inline client/src/components/ItemSummary.jsx]: You might consider getting out of the habit of using abbreviated terms like "Obj", at least for now until you're in an environment where it's the norm.

File: client/src/components/ItemSummary.jsx [hunk 4]
```diff
@@ -36,34 +36,37 @@ class ItemSummary extends React.Component {
...
 
   render() {
+    var item = this.state.item;
     return (
       <Main>
         <NameAndPrice
-          name={this.state.item.length ? this.state.item[0].name : null}
-          price={this.state.item.length ? this.state.item[0].price : null}
+          name={item.length ? item[0].name : null}
+          price={item.length ? item[0].price : null}
```
Comment (garrowat) [inline client/src/components/ItemSummary.jsx]: Nice use of ternaries. 💯

File: client/src/components/SizeDropDown.jsx [hunk 5]
```diff
@@ -60,7 +60,11 @@ class SizeDropDown extends React.Component {
         <DropDownHeader selectedSize={this.state.selectedSize} toggleDropDown={this.toggleDropDown}/>
         <DropDownList dropDownOpen={this.state.dropDownOpen}>
           {this.props.sizes.map((size) => {
-            return <DropDownItem key={size} value={size} onClick={this.handleDropDownSubmit}>{size}</DropDownItem>
+            if (size.stock === 0) {
+              return <DropDownItem key={size.size} value={size.size} onClick={this.handleDropDownSubmit}>{size.size}(out of stock)</DropDownItem>
```
Comment (garrowat) [inline client/src/components/SizeDropDown.jsx]: size.size is a little odd sounding here, but looks good to me otherwise.

Comment (matt-leyrat) [reply client/src/components/SizeDropDown.jsx]: The size object has two properties: size and stock. I couldn't think of a good name at the time, maybe size.sizeCode or itemSize.size? 

The size object is also nested inside a color object, so I have to keep that context in mind when naming these properties.

color.itemSize.size vs. color.size.sizeCode, idk which one will make it easier to read in the future.

Comment (garrowat) [reply client/src/components/SizeDropDown.jsx]: Yeah that's tough, maybe the first object could be named something like "sizeInfo"?

File: client/src/components/AccordionTab.jsx (hunk 1 above)
Comment (mchung1128) [reply client/src/components/AccordionTab.jsx]: Yes, not much to add here, but agree to Garrett!

File: server/seed.js [hunk 6]
```diff
@@ -13,34 +22,33 @@ const descriptions = [
...
 ];
 
-const fabrics = ['Wool', 'Calico', 'Silk', 'Camel Hair', 'Hemp', 'Canvas', 'Cashmere', 'Linen', 'Velvet'];
+// const fabrics = ['Wool', 'Calico', 'Silk', 'Camel Hair', 'Hemp', 'Canvas', 'Cashmere', 'Linen', 'Velvet'];
+
+const fabrics = [
```
Comment (mchung1128) [inline server/seed.js]: Ohh this is really cool, I'm excited to see how this looks now!

File: client/src/components/Accordions.jsx [hunk 7]
```diff
@@ -56,7 +55,9 @@ class Accordion extends React.Component {
               tabId={tab.id}
               openedTab={this.state.openedTab}
               care={this.props.care}
-              designedFor={this.props.designedFor}
+              fabric={this.props.fabric}
```
Comment (mchung1128) [inline client/src/components/Accordions.jsx]: Same as last comment, use of destructuring will make this a bit cleaner!

File: client/src/components/SizeDropDown.jsx (hunk 5 above)
Comment (mchung1128) [reply client/src/components/SizeDropDown.jsx]: I agree here, I actually did a double take when I saw "size.size". I think sizeInfo will make things a little less confusing.

**Pull request merged**